src/librustc_target/abi/mod.rs

```diff
@@ -453,6 +452,298 @@ impl AbiAndPrefAlign {
     }
 }
 
+/// An aligned size preference.
+/// Better name appreciated.
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug, RustcEncodable, RustcDecodable)]
+pub struct LayoutPositionPref {
```

Comment:
"Position" is just wrong, isn't it? And "pref" clashes with "preferred alignment".

Reply:
"Position" seemed fine to me: "a particular way in which someone or something is placed or arranged" and/or "where something should be". As this structure is describing where an object should be, relative to the alignment of memory, it seems to fit.

And "Pref" was originally because of "preferred alignment", in my first drafts it was "MemPosPref" and the other was "MemPos". I changed the name when I realized it was (almost) entirely used only in the `::layout` module.
 
If you have a better idea, I can change it, naming things is hard. 🤷‍♂️

Reply:
But it's a measure of how big something is, not where to put it.
I'm not sure what you mean by "relative to the alignment".

Okay if you want to use Pref to refer to the fact that you use `AbiAndPrefAlign`, you'd need a much longer name. Or a single type generic on the type of the align field.

See my top-level comment though.